spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/test/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfigurationTests.java

```diff
@@ -55,13 +59,46 @@ public void testGetProjectIdProvider_withGcpProperties() {
 		assertEquals("test-project", provider.getProjectId());
 	}
 
+	@Test
+	public void testStarterScopes() {
+	}
+
 	@Test
 	public void testGetProjectIdProvider_withoutGcpProperties() {
 		loadEnvironment();
 		assertTrue(this.context.getBean(GcpProjectIdProvider.class)
 				instanceof DefaultGcpProjectIdProvider);
 	}
 
+	@Test
+	public void testResolveScopesDefaultScopes() {
+		loadEnvironment();
+		GcpContextAutoConfiguration configuration = this.context.getBean(GcpContextAutoConfiguration.class);
+		List<String> scopes = configuration.resolveScopes();
+		assertTrue(scopes.size() > 1);
+		assertTrue(scopes.contains(GcpScope.PUBSUB.getUrl()));
+	}
+
+	@Test
+	public void testResolveScopesOverrideScopes() {
+		loadEnvironment("spring.cloud.gcp.credentials.scopes=myscope");
+		GcpContextAutoConfiguration configuration = this.context.getBean(GcpContextAutoConfiguration.class);
+		List<String> scopes = configuration.resolveScopes();
+		assertEquals(scopes.size(), 1);
+		assertTrue(scopes.contains("myscope"));
+	}
+
+	@Test
+	public void testResolveScopesStarterScopesPlaceholder() {
+		loadEnvironment("spring.cloud.gcp.credentials.scopes=STARTER_SCOPES,myscope");
+		GcpContextAutoConfiguration configuration = this.context.getBean(GcpContextAutoConfiguration.class);
+		List<String> scopes = configuration.resolveScopes();
+		assertTrue(scopes.size() > 1);
```

Comment:
We could test the exact number of scopes. Testing if there is more than one scope isn't extremely useful.